### PR TITLE
feat: rootless Podman with documentation

### LINKED ISSUE #2178: [FR] document how to run in rootless Podman
# Feature Request

This is more or less a follow-up of @OrvilleQ's [PR that brought general Podman support](https://github.com/docker-mailserver/docker-mailserver/pull/1327).
[As OrvilleQ mentioned earlier](https://github.com/docker-mailserver/docker-mailserver/pull/1327#issuecomment-557875369), another PR should fix this.

## Context

Currently [the script will complain](https://github.com/docker-mailserver/docker-mailserver/blob/0da66ccb34dcaf0663b6b5635ba97f990b761881/setup.sh#L73) with the message `Curently docker-mailserver doesn't support podman's rootless mode` if it is running rootless.

### Is your Feature Request related to a Problem?

docker-mailserver cannot be run in rootless mode.

### Describe the Solution you'd like

It should be running in rootless mode.

I'm new to podman, so I don't know exactly how a proper solution should look like.
To be honest, I'm not even sure what it needs root for.
As far as I understood, the port within the container image shouldn't matter (since it can be mapped at the host), right?
And on the outside only the `CAP_NET_BIND_SERVICE` capability should be needed for privileged ports, right?

### Are you going to implement it?

No, and I understand that it is highly likely no one will implement it. Furthermore, I understand that this issue will likely become stale and will be closed.

### Alternatives you've considered

Not using `docker-mailserver` for projects that require rootless mode, I guess? :shrug: 

### Who will that Feature be useful to?

For people using rootless services as it is commonly suggested.

### What have you done already?

Nothing except writing this issue.

### PR DESCRIPTION
**Pull request by OrvilleQ**
# Description

Setup.sh : Add ability to operate rootless podman.

Doc: Add guide for Podman and Root-less Podman

Fixes #2178

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)
- [x] This change requires a documentation update

## Checklist:

- [x]  My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by georglauterbach** [comment]
I will mark this as WIP as long as #2174 is not merged because #2174 is important. You will then likely need to rebase onto `master`. Reviews can then be provided. In the meantime, I'd like to discuss this topic in the linked issue #2178.

File: docs/content/config/advanced/podman.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    This guide only tested under Fedora 34 with systemd and firewalld. Also it require podman version >= 3.2.
+
+## Root Mode
+
+while using podman, you can just manage docker-mailserver as what you did with docker. Your best friend setup.sh include the minimum code in order to support podman since it's 100% compatible with Docker CLI.
+
+And for the installation is also basically the same. Podman v3.2 introduced a RESTful API that is 100% compatible with Docker API, so you could just use docker-compose with podman without pain.
+
+### Installnation
```
Comment (alexanderadam) [inline docs/content/config/advanced/podman.md:25]: `s/Installnation/Installation`

File: setup.sh [hunk 2]
```diff
@@ -70,8 +70,12 @@ function _check_root
...
   if [[ ${EUID} -ne 0 ]]
   then
-    echo "Curently docker-mailserver doesn't support podman's rootless mode, please run this script as root user."
-    exit 1
+    read -p "You are now running in the rootless mode of podman, are you sure you want to continue? (Y/N):" -r yn
+    case ${yn} in
+      [Yy]* ) ;;
+      [Nn]* ) exit 1;;
+      * ) echo "Please answer Y(es) or N(o)."; exit 1;;
+    esac
```
Comment (georglauterbach) [inline setup.sh:78]: Please use this suggestion to follow general `read` conventions (chose a default value, make use of the `${REPLY}` variable, etc.). Make sure to adhere to the style conventions (always uppercase variables) too :)

``` SUGGESTION
    read -r -p "You are now running in the rootless mode of podman, are you sure you want to continue? [Y/n] "
    if [[ -n ${REPLY} ]] && [[ ${REPLY} =~ (n|N) ]]
      exit 0
   fi
```

**Comment by georglauterbach** [comment]
I have adjusted to documentation :) @polarathene what do you think of this PR?

**Review by wernerfred — APPROVED** [review]
LGTM 📚 👍🏻

**Comment by georglauterbach** [comment]
@wernerfred I messed up `setup.sh` for a second with one of my commits. Please review again :)

**Review by georglauterbach — DISMISSED** [review]

**Comment by georglauterbach** [comment]
**NOTE** Tests for merge requests are currently expected to fail. When #2189 is merged, tests should pass again.

**Review by wernerfred — APPROVED** [review]

**Comment by casperklein** [comment]
`setup.sh` is non-interactive and all configuration is based on parameters. Why switching to interactive for podman?

This makes it harder, if you use setup.sh in an automated way. There should be an "rootless" option, when calling `setup.sh`.

**Comment by georglauterbach** [comment]
> `setup.sh` is non-interactive and all configuration is based on parameters. Why switching to interactive for podman?
> 
> This makes it harder, if you use setup.sh in an automated way. There should be an "rootless" option, when calling `setup.sh`.

Agreed. I thought about exactly the same when doing the change. But should be done with another PR :)

**Comment by georglauterbach** [comment]
Unexpected test failure:

``` BASH
 not ok 296 checking system: postfix should not log to syslog
# (from function `assert_failure' in file test/test_helper/bats-assert/src/assert.bash, line 140,
#  in test file test/tests.bats, line 549)
#   `assert_failure' failed
# 
# -- command succeeded, but it was expected to fail --
# output : Sep 16 09:33:13 mail postfix/postfix-script[2669]: fatal: the Postfix mail system is already running
# --
# 
```

Re-run the tests?

**Comment by wernerfred** [comment]
> But should be done with another PR :)

Why not change it now?

**Comment by georglauterbach** [comment]
@wernerfred I'd rather like to have a look at the whole "argument" parsing TBH. Therefore, I'd like to see the functionality being preserved with this PR and change with a follow-up.

**Comment by georglauterbach** [comment]
I have completely removed the changes to `setup.sh` as I will provide a dedicated PR for that.

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**